### PR TITLE
email address is present, valid, and unique

### PR DESCRIPTION
**Pull request by NataMarkova**
@jrmcgarvey , please check it out. This PR covers the ticket "The user class does not ensure that the email address is present and valid and unique."

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -6,6 +6,9 @@ class User < ApplicationRecord
   devise :database_authenticatable, :registerable,      
          :recoverable, :rememberable, :validatable, password_length: 8..25
   validate :password_complexity
+  validates :email, presence: true
+  validates_format_of :email, :with => /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\Z/i
+  validates :email, uniqueness: true
```
Comment (jrmcgarvey) [inline app/models/user.rb:11]: line 11 should be:
validates :email, uniqueness: true, case_sensitive: false

**Review by jrmcgarvey — CHANGES_REQUESTED** [review]
I put one comment in line, but the case handling for the email should be as described here: https://stackoverflow.com/questions/4148133/rails-devise-email-case-insensitivity

File: app/models/user.rb (hunk 1 above)
Comment (NataMarkova) [reply app/models/user.rb]: Added this piece of code

**Review by jrmcgarvey — APPROVED** [review]
ok!

**Pull request merged**